src/token/src/canister.rs

```diff
@@ -293,11 +331,14 @@ impl TokenCanister {
     /// If owner calls this method and `from` is Some(who), then who's tokens will be burned.
     #[update]
     fn burn(&self, from: Option<Principal>, amount: Nat) -> TxReceipt {
-        if from.is_some() {
-            check_caller(self.owner())?;
+        match from {
+            None => burn_own_tokens(self, amount),
+            Some(from) if from == ic_kit::ic::caller() => burn_own_tokens(self, amount),
```

Comment:
Don't you want to make this also a `CheckedPrincipal`? Because `burn_own_tokens` functions by itself looks dangerous at the moment.

Reply:
From our Slack conversation it seems like all is well so I'll resolve this